.clang-tidy

```diff
@@ -32,7 +32,8 @@ Checks:
     portability-std-allocator-const,
     portability-simd-intrinsics,
     readability-*,
-    -readability-avoid-const-params-in-decls
+    -readability-avoid-const-params-in-decls,
+    -readability-function-cognitive-complexity
```

Comment:
I would prefer to leave this check enabled, but as a warning rather than an error

Reply:
I'm not sure this check offers much to us, but I'm willing to be convinced.

Reply:
After discussion, the plan is to look at configuring this option and seeing how it behaves in the future.

Reply:
The more I think about this one, the more I feel keeping it is the right thing to do. It's only an issue when looking at old C code, which means it is likely doing its job.

Reply:
this seems very reasonable.    can always change it later if you still find you don't like it.

Reply:
I agree. Closing #6594 as a result of these conversations.

Reply:
Done.